### PR TITLE
gives fish analyzers a UI

### PR DESCRIPTION
**Pull request by Ben10Omintrix**
## About The Pull Request
makes fish analyzers use a UI instead of the radial system, which makes them a bit easier to use 
fishscreenie

## Why It's Good For The Game
makes fish analyzers easier and more convenient to use.

## Changelog
:cl:
qol: fish analyzers now have a UI
/:cl:

### CODE REVIEW

**Comment by MTandi** [comment]
Please choose one of the existing themes instead of some random linear gradient. It will look a lot better.

**Comment by Ghommie** [comment]
Honestly, I prefer this gradient. It has soul. PS1 levels of soul.

File: tgui/packages/tgui/interfaces/FishAnalyzer.tsx [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+  Flex,
+  Icon,
+  Input,
+  ProgressBar,
+  Section,
+  Stack,
+} from '../components';
+import { Window } from '../layouts';
+
+type fishData = {
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/FishAnalyzer.tsx:19]: types are pascal case pls

File: tgui/packages/tgui/interfaces/FishAnalyzer.tsx [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+export const FishAnalyzer = (props) => {
+  const { act, data } = useBackend<Data>();
+  const { fish_list = [], fish_scanned } = data;
+  const [searchText, setSearchText] = useState('');
+
+  const search = createSearch(searchText, (fish: fishData) => fish.fish_name);
+
+  const fish_filtered =
+    searchText.length > 0
+      ? fish_list.filter((fish) => search(fish))
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/FishAnalyzer.tsx:67]:
```suggestion
      ? fish_list.filter(search)
```

File: tgui/packages/tgui/interfaces/FishAnalyzer.tsx [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+  if (input.length > 12) {
+    const firstSpace = input.indexOf(' ');
+
+    if (!firstSpace) {
+      return input.substring(0, 12);
+    }
+    return input.substring(0, firstSpace);
+  }
+  return input;
+}
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/FishAnalyzer.tsx:332]: could just simplify this to
```suggestion
function dissectName(input: string): string {
  return input.split(' ').slice(0, 12);
}
```
also, functions are camelCase. only types and components get pascal case like this

File: tgui/packages/tgui/interfaces/FishAnalyzer.tsx [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+  evolution_probability: number;
+  evolution_conditions: string;
+};
+
+type Data = {
+  fish_list: fishData[];
+  fish_scanned: BooleanLike;
+};
+
+export const FishAnalyzer = (props) => {
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/FishAnalyzer.tsx:58]: this component is pretty huge, perhaps could be split into subcomponents at each `.map`

ideal components are between 30-100 lines (loose)

**Comment by jlsnow301** [comment]
loose approval on this terrible looking gradient for this very reason

> PS1 levels of soul

**Comment by Ben10Omintrix** [comment]
tried looking for some better aquatic themed gradients hopefully you like this better 

gradd

**Comment by jlsnow301** [comment]
> tried looking for some better aquatic themed gradients hopefully you like this better

i unironically like the first better

**Comment by Ben10Omintrix** [comment]
OK ill just be using a set shade of purple rather than a lineargradient then

**Comment by san7890** [comment]
oeugh tsx

**Pull request merged**